src/bundles/physics_2d/PhysicsObject.ts

```diff
@@ -0,0 +1,192 @@
+    this.body.SetAngle(rot);
+  }
+
+  public getVelocity() {
+    return this.body.GetLinearVelocity();
+  }
+
+  public setVelocity(velc: b2Vec2) {
+    this.body.SetLinearVelocity(velc);
+  }
+
+  public getAngularVelocity() {
+    return this.body.GetAngularVelocity();
+  }
+
+  public setAngularVelocity(velc: number) {
+    this.body.SetAngularVelocity(velc);
+  }
+
+  public addForceCentered(force: Force) {
+    this.forcesCentered.push(force);
+  }
+
+  public addForceAtAPoint(force: Force, pos: b2Vec2) {
+    this.forcesAtAPoint.push({
+      force,
+      pos,
+    });
+  }
+
+  private applyForcesToCenter(world_time: number) {
+    this.forcesCentered = this.forcesCentered.filter(
+      (force: Force) => force.start_time + force.duration > world_time,
+    );
+
+    const resForce = this.forcesCentered
+      .filter((force: Force) => force.start_time < world_time)
+      .reduce(
+        (res: b2Vec2, force: Force) => res.Add(force.direction.Scale(force.magnitude)),
+        new b2Vec2(),
+      );
+
+    this.body.ApplyForceToCenter(resForce);
+  }
+
+  private applyForcesAtAPoint(world_time: number) {
+    this.forcesAtAPoint = this.forcesAtAPoint.filter(
+      (forceWithPos: ForceWithPos) => forceWithPos.force.start_time + forceWithPos.force.duration > world_time,
+    );
+
+    this.forcesAtAPoint.forEach((forceWithPos) => {
+      const force = forceWithPos.force;
+      this.body.ApplyForce(
+        force.direction.Scale(force.magnitude),
+        forceWithPos.pos,
+      );
+    });
+  }
+
+  public applyForces(world_time: number) {
+    this.applyForcesToCenter(world_time);
+    this.applyForcesAtAPoint(world_time);
+  }
+
+  public isTouching(obj2: PhysicsObject) {
+    let ce = this.body.GetContactList();
+    while (ce !== null) {
+      if (ce.other === obj2.body && ce.contact.IsTouching()) {
+        return true;
+      }
+      ce = ce.next;
+    }
+    return false;
+  }
+
+  public toReplString = () => `
+  Mass: ${this.getMass()
+    .toFixed(ACCURACY)}
+  Position: [${this.getPosition().x.toFixed(
+    ACCURACY,
+  )},${this.getPosition().y.toFixed(ACCURACY)}]
+  Velocity: [${this.getVelocity().x.toFixed(
+    ACCURACY,
+  )},${this.getVelocity().y.toFixed(ACCURACY)}] 
+  
+  Rotation: ${this.getRotation()
+    .toFixed(ACCURACY)}
+  AngularVelocity: [${this.getAngularVelocity()
+    .toFixed(ACCURACY)}]`;
+
+  public scale_size(scale: number) {
+    if (this.shape instanceof b2CircleShape) {
+      this.shape.m_radius *= scale;
+    } else if (this.shape instanceof b2PolygonShape) {
+      let centroid: b2Vec2 = this.shape.m_centroid;
+      let arr: b2Vec2[] = [];
+      this.shape.m_vertices.forEach((vec) => {
+        arr.push(
+          new b2Vec2(
+            centroid.x + scale * (vec.x - centroid.x),
+            centroid.y + scale * (vec.y - centroid.y),
+          ),
+        );
+      });
+      this.shape = new b2PolygonShape()
+        .Set(arr);
+    }
+    let f: b2Fixture = this.fixture;

```

Comment:
Is there any reason that `centroid`, `arr` and `f` are declared using the `let` keyword instead of `const`?

Reply:
Nope. Will fix and create a new PR with these changes.